### DISCUSSION TITLE
Partial "labels to fields" transformation?

### DISCUSSION THREAD

**Discussion by tudor** [grafana/grafana#46713] [Dashboards, Panels & Visualizations]
I have a Prometheus time series of the following form

```
Time = 100, status_info{id="foo", state="ok"} = 1
Time = 100, status_info{id="bar", state="in progress"} = 1
Time = 101, status_info{id="foo", state="error"} = 1
Time = 101, status_info{id="bar", state="in progress"} = 1
```

etc.

(The actual content is a bunch of key/value strings, but because Prometheus only supports numeric values, it is represented by adding the key/value strings as labels and setting the value to 1; this is what the Info metric type in Prometheus client libraries does.)

I'd like to represent this in Grafana using the "State timeline" panel, with one row (horizontal bar) for every unique `id` label.

The "labels to fields" transformation allows me to extract the `state` label into a field, but it loses the `id` label on the data; I'd like to just extract the `state` label and keep the `id` label.

That is, I'd like the output of the transformation to be

```
Time = 100, state{id="foo"} = "ok"
Time = 100, state{id="bar"} = "in progress"
Time = 101, state{id="foo"} = "error"
Time = 101, state{id="bar"} = "in progress"
```

Is doing something like this possible?

**Comment by tudor** [comment]
I do control the source of the data, so I *could* map all the possible states to integers and map them back to strings in the dashboard, but that's cumbersome and error-prone (there are about 15 possible states).

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
